src/odemis/driver/tmcm.py

```diff
@@ -427,22 +419,39 @@ def __init__(self, name, role, port, axes, ustepsize, address=None,
             self._updateTemperatureVA() # make sure the temperature is correct
             self._temp_timer.start()
 
-    # def _is_already_ref(self, an):
-    #     if self._abs_encoder[an] is True:
-    #         return True
-    #     # Check the user variable is in the "right format"
-    #     self.aref
-    #     # if not -> reset UVXXX to 0, and return False
-    #     gparam = 117
-    #     self.SetGlobalParam(2, gparam, 0)
-    #     return F
-    #     # return UVXXX value at [an]
-    #     an = self.aref
-    #     return an
-
-    def _set_ref(self, an, value):
-        # read the curren value of UVXXX
-        #update UVXXX with an = value
+    def _is_already_ref(self, an):
+        """
+
+        Args:
+            an: axis number
+
+        Returns (boolean): if the axis is referenced or not
+
+        """
+        if self._abs_encoder[an] is True:
+            return True
+        arefd = self.GetGlobalParam(2, 117)
+        # Is arefd the right format? if not -> return False
+        if arefd & 0xffff == ~(arefd >>16):
+            aref_an = 1 << an
+            return bool(arefd & aref_an)
+        # if not reset the user variable and return False
+        else:
```

Comment:
Only reset it if it's not 0 already. Otherwise it feels a little over-the-top to keep writing 0 in this variable for every axis.